### PR TITLE
feat(badge): add notification variant

### PR DESCRIPTION
**Pull request by krisantrobus**
This pull request primarily introduces a new badge variant, `notification_counter`, to the badge component in `packages/paste-core/components/badge`. 

It uses the new design tokens for notifications

### CODE REVIEW

File: packages/paste-website/src/pages/components/badge/index.mdx [hunk 1]
```diff
@@ -405,6 +405,19 @@ Use a small Badge only when vertical density is a concern. The guidelines for us
 `.trim()}
 </LivePreview>
 
+### Notification Counter
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/badge/index.mdx:408]:
```suggestion
### Notification counter
```

[the remark this answers is not in the document]
Comment (serifluous) [reply packages/paste-website/src/pages/components/badge/index.mdx:408]: Also, can you replace the text in the "Counter" section with this? lol noticed more typos:

> Use the counter Badge to visually highlight a count in a UI, like the number of pending invitations or the number of errors. Counter Badges are limited to neutral and error variants, and the error variant should always include an error icon.

[the remark this answers is not in the document]
Comment (krisantrobus) [reply packages/paste-website/src/pages/components/badge/index.mdx:408]: I like it, added to upcoming commit

**Review by nkrantz — DISMISSED** [review]

**Comment by roxanagomez** [comment]
TODO/nit-pick: add notification badge example to first code example of badges. 
Screenshot 2024-07-18 at 9 40 11 AM

**Review by serifluous — APPROVED** [review]

**Review by roxanagomez — APPROVED** [review]

**Review by nkrantz — APPROVED** [review]

**Pull request merged**